src/os/bluestore/BlueStore.cc

```diff
@@ -13623,6 +13691,15 @@ int BlueStore::_do_alloc_write(
   }
   _collect_allocation_stats(need, min_alloc_size, prealloc.size());
 
+  if (bdev->is_smr()) {
+    std::lock_guard l{zoned_cleaner_lock};
+    if (!zoned_cleaner_in_progress && shared_alloc.a->zoned_needs_cleaning()) {
+      zoned_cleaner_queue = shared_alloc.a->zoned_get_zones_to_clean();
```

Comment:
You can get rid off zoned_cleaner_in_progress if you APPEND (can be done optionally if zoned_cleaner_queue isn't empty) content to zoned_cleaner_queue and do move its content to local queue instance in the thread before the processing as per my comment above.

Reply:
Done.